hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionPlacementMaintainer.java

```diff
@@ -204,9 +202,7 @@ private void genAssignmentPlan(TableName tableName,
 
     // Get the all the region servers
     List<ServerName> servers = new ArrayList<>();
-    servers.addAll(
-      FutureUtils.get(getConnection().getAdmin().getClusterMetrics(EnumSet.of(Option.LIVE_SERVERS)))
-        .getLiveServerMetrics().keySet());
+    servers.addAll(FutureUtils.get(getConnection().getAdmin().getRegionServers()));
```

Comment:
Not introduced by you, but I wonder whether we need to close the Admin here?

Reply:
It's a async admin and does not need to close.